mobile/android/android-emulator-services.ts

```diff
@@ -120,15 +152,124 @@ class AndroidEmulatorServices implements Mobile.IEmulatorPlatformServices {
 		return timeout * 1000;
 	}
 
+	private listVirtualMachines(parameter: string): IFuture<VirtualMachine[]> {
+		var future = new Future<VirtualMachine[]>();
+		var allData: string = "";
+		var virtualMachines: VirtualMachine[] = [];
+
+		var vboxPath = path.join(this.getPathToVBox().wait(), "vboxmanage");
+		var vboxManage = this.$childProcess.spawn(vboxPath, ["list", parameter]);
+
+		vboxManage.stdout.on("data", (chunk: string) => {
+			if(chunk) {
+				allData += chunk;
+			}
+		});
+		vboxManage.stderr.on("data", (error: Error) => {
+			// maybe use log trace level to print the error
+			this.$errors.fail("Unable to get currently running virtual machines. Check if VirtualBox is installed correctly.");
+		});
+
+		vboxManage.on("exit", () => {
+			_(allData.split(os.EOL)).each(row => {
+				// sample output row: "GN5" {1c268a7b-aa04-49c9-aabd-90595a747dec}
+				var rx = row.match(/^"(.+)"\s+\{(.+)\}$/);
+				if(rx && rx[1]) {
+					virtualMachines.push(new VirtualMachine(rx[1], rx[2]));
+				}
+			});
+
+			future.return(virtualMachines);
+		});
+
+		return future;
+	}
+
+	private shouldStartGenymotionEmulator(image: string): IFuture<boolean> {
+		return (() => {
+			if(!options.geny) {
+				return false;
+			}
+
+			var hasVmWithSpeciedImage = false;
+			var allVMs: VirtualMachine[] = this.listVirtualMachines("vms").wait();
+			_(allVMs).each((vm: VirtualMachine) => {
+				if(vm.name === image || vm.identifier === image) {
+					hasVmWithSpeciedImage = true;
+					// break each
+					return false;
+				}
+			});
+
+			if(!hasVmWithSpeciedImage) {
+				this.$errors.fail("There is no VirtualBox machine with name %s. Please verify specified name is correct.", image);
+			}
+
+			var shouldStartEmulator = true;
+			var virtualMachines: VirtualMachine[] = this.listVirtualMachines("runningvms").wait();
+			_(virtualMachines).each((vm: VirtualMachine) => {
+				if(vm.name === image || vm.identifier === image) {
+					shouldStartEmulator = false;
+					// break each
+					return false;
+				}
+			});
+
+			return shouldStartEmulator;
+		}).future<boolean>()();
+	}
+
+	private getPathToVBox(): IFuture<string> {
+		if(hostInfo.isWindows()) {
+			var result = new Future<string>();
+			var regKey64 = new Winreg({
+				hive: Winreg.HKLM,
+				key: '\\SOFTWARE\\Oracle\\VirtualBox',
+				regVersion: '64'
+			});
+
+			var regKey32 = new Winreg({
+				hive: Winreg.HKLM,
+				key: '\\SOFTWARE\\Oracle\\VirtualBox',
+				regVersion: '32'
+			});
+
+			regKey64.get("InstallDir", (err: Error, item: any) => {
+				if(err) {
+					regKey32.get("InstallDir", (err: Error, item: any) => {
+						if(err) {
+							this.$errors.fail({ formatStr: "Make sure VirtualBox is installed correctly.", suppressCommandHelp: true });
```

Comment:
Maybe if we fail with regKey64 and regKey32, we should try getting the value without passing regVersion. I'm not sure for which versions of REG tool /reg:32 (/reg:64) option are supported. If there's a system where we fail when passing them, maybe the fail is because of the usage of this parameters. So we can try executing the command without this parameter and see if there's a value in the registry.